misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -96,135 +97,149 @@
+                    ModuleID lftModule = lftTypeSymbol.moduleID();
+
+                    if (!moduleName.equals(rhsModule.moduleName())) {
+                        String rhsType = rhsModule.modulePrefix() + ":" + foundTypeRight;
+                        generatedRecordMappingFunction = generatedRecordMappingFunction.replaceAll("\\b" +
+                                foundTypeRight + "\\b", rhsType);
+                    }
+                    if (!moduleName.equals(lftModule.moduleName())) {
+                        String lftType = lftModule.modulePrefix() + ":" + foundTypeLeft;
+                        generatedRecordMappingFunction = generatedRecordMappingFunction.replaceAll("\\b" +
+                                foundTypeLeft + "\\b", lftType);
+                    }
                 }
-                break;
-            default:
-                throw new IllegalStateException("Unexpected value: " + symbolAtCursorType);
-        }
 
-        // Insert function declaration at the bottom of the file
-        String functionName = String.format("map%sTo%s", foundTypeRight, foundTypeLeft);
-
-        boolean found = fileContentSymbols.stream().anyMatch(p -> p.name().contains(functionName));
-        if (!found) {
-            TextDocument fileContentTextDocument = context.workspace().syntaxTree(context.filePath()).get().
-                    textDocument();
-            int numberOfLinesInFile = fileContentTextDocument.toString().split("\n").length;
-            Position startPosOfLastLine = new Position(numberOfLinesInFile + 2, 0);
-            Position endPosOfLastLine = new Position(numberOfLinesInFile + 2, 1);
-            Range newFunctionRange = new Range(startPosOfLastLine, endPosOfLastLine);
-            String generatedRecordMappingFunction =
-                    getGeneratedRecordMappingFunction(context.positionDetails(), context, foundTypeLeft,
-                            foundTypeRight, fileContentSymbols);
-            fEdits.add(new TextEdit(newFunctionRange, generatedRecordMappingFunction));
+                fEdits.add(new TextEdit(newFunctionRange, generatedRecordMappingFunction));
+            }
+            return fEdits;
         }
-        return fEdits;
     }
 
     /**
      * Check if a symbol is in the module level, and if so return the symbol.
      *
-     * @param symbolName         {@link String}
      * @param fileContentSymbols {@link List}
      * @return return the symbol
      */
-    public static Symbol findSymbol(String symbolName, List<Symbol> fileContentSymbols) {
-        Iterator<Symbol> iterator = fileContentSymbols.iterator();
-        while (iterator.hasNext()) {
-            Symbol symbol = iterator.next();
-            if (symbol.name().equals(symbolName)) {
+    public static Symbol findSymbol(List<TypeSymbol> fileContentSymbols) {
+        for (Symbol symbol : fileContentSymbols) {
+            if (!symbol.name().equals("ERROR")) {
```

Comment:
This should be otherway, to avoid possible NPEs. `"ERROR".equals(symbol.name())`

Reply:
Modified the code according to the comment